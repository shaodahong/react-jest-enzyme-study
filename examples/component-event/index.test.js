import React from 'react';
import ComponentEvent from './index';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, render, mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('ComponentEvent test', () => {
  it('ComponentEvent test', () => {
    const wrapper = shallow(<ComponentEvent />);
    expect(wrapper).toMatchSnapshot();
  });

  it('ComponentEvent Button Click', () => {
    const wrapper = shallow(<ComponentEvent />);

    const onClickButtonMock = jest.spyOn(wrapper.instance(), 'onClickButton');

    wrapper.instance().forceUpdate();

    expect(onClickButtonMock).toHaveBeenCalledTimes(0);

    wrapper.find('button').prop('onClick')();

    expect(onClickButtonMock).toHaveBeenCalledTimes(1);
  });

  it('ComponentEvent Input Change', () => {
    const wrapper = shallow(<ComponentEvent />);

    wrapper.find('input').prop('onChange')({
      target: {
        value: 'mock'
      }
    });

    expect(wrapper.find('input').prop('value')).toBe('mock');
  });
});
