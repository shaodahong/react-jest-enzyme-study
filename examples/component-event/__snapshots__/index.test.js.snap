// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentEvent test ComponentEvent test 1`] = `
<div>
  <button
    onClick={[Function]}
  />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
